### PR TITLE
Clean up n2o_component and other problems

### PR DESCRIPTION
**Pull request by bpbond**
Clean up `n2o_component.cpp` and related data structures, which had confusing notation, unused variables, and incorrect internal units (although its outputs were correct). Also fixed a few minor problems noted by clang and updated Xcode project file settings.

### CODE REVIEW

File: source/components/n2o_component.cpp [hunk 1]
```diff
@@ -133,27 +133,29 @@ void N2OComponent::run( const double runToDate ) throw ( h_exception ) {
...
     
-    //varying natural emissions from preindustrial to 2000.  After 2000, we hold natural emissions constant
-       
-    const double current_n2oem = N2O_emissions.get( runToDate ).value( U_TG_N2O ) + N2ON_emissions.get( runToDate ).value( U_TG_N2O ); 
+    // Current emissions are the sum of natural and anthropogenic sources
+    const double current_n2oem = N2O_emissions.get( runToDate ).value( U_TG_N ) + N2O_natural_emissions.get( runToDate ).value( U_TG_N );
     
-    double dN2O = current_n2oem/UC_N2O - previous_n2o/TAU_N2O.get( runToDate ).value( U_YRS );
+    // This calculation follows Eq. B7 in Ward and Mahowald 2014
+    const double dN2O = current_n2oem / UC_N2O - previous_n2o / TAU_N2O.get( runToDate ).value( U_UNITLESS );
```
Comment (rplzzz) [inline source/components/n2o_component.cpp]: Is `TAU_N2O` really unitless?  It seems like the kind of thing that should be in years.  Indeed, it was in years before.  Why the change?

File: headers/components/ch4_component.hpp [hunk 2]
```diff
@@ -62,11 +60,11 @@ class CH4Component : public IModelComponent {
      //! emissions time series
     tseries<unitval> CH4_emissions;
     tseries<unitval> CH4;  // CH4 concentrations
-    unitval M0;     // preindustrial methane
+    unitval M0;     // preindustrial methane, Tg
```
[the remark this answers is not in the document]
Comment (bpbond) [reply headers/components/ch4_component.hpp]: Whoops, thank you!

File: headers/components/ch4_component.hpp [hunk 3]
```diff
@@ -62,11 +60,11 @@ class CH4Component : public IModelComponent {
...
     unitval UC_CH4;  // conversion factor between emissions and concentration
-    unitval CH4N;   // natural emissions
-    unitval Tsoil;  // CH4 loss to soil
-    unitval Tstrat; //  CH4 loss to stratosphere
+    unitval CH4N;   // natural emissions, Tg
```
[the remark this answers is not in the document]
Comment (bpbond) [reply headers/components/ch4_component.hpp]: At the very least this can be clarified in the comment. Which I'll do.

File: source/components/n2o_component.cpp [hunk 4]
```diff
@@ -96,19 +96,19 @@ void N2OComponent::setData( const string& varName,
...
-            N2ON_emissions.set(data.date, data.getUnitval(U_TG_N2O));
-        } else if( varName == D_CONVERSION_N2O ) {
+            N2O_natural_emissions.set(data.date, data.getUnitval( U_TG_N ) );
+        }  else if( varName == D_CONVERSION_N2O ) {
             H_ASSERT( data.date == Core::undefinedIndex(), "date not allowed" );
             UC_N2O = data.getUnitval(U_TG_PPBV);
         } else if( varName == D_INITIAL_LIFETIME_N2O ) {
             H_ASSERT( data.date == Core::undefinedIndex(), "date not allowed" );
-            TN2O0 = data.getUnitval(U_YRS);
+            TN2O0 = data.getUnitval( U_YRS );
```
[the remark this answers is not in the document]
Comment (bpbond) [reply source/components/n2o_component.cpp]: I was just trying for consistency. Most of the code follows the GCAM convention (which I don't like...).

File: source/components/ch4_component.cpp [hunk 5]
```diff
@@ -58,16 +57,15 @@ void CH4Component::init( Core* coreptr ) {
...
     core->registerInput(D_EMISSIONS_CH4, getComponentName());
 }
 
 //------------------------------------------------------------------------------
 // documentation is inherited
 unitval CH4Component::sendMessage( const std::string& message,
                                   const std::string& datum,
-                                  const message_data info ) throw ( h_exception )
-{
+                                  const message_data info ) throw ( h_exception ) {
```
[the remark this answers is not in the document]
Comment (bpbond) [reply source/components/ch4_component.cpp]: Opened #257

File: source/components/n2o_component.cpp (hunk 1 above)
Comment (bpbond) [reply source/components/n2o_component.cpp]: Mmm...looking back at the at the Ward et al. equation, I think you're right. Thanks.

File: source/components/n2o_component.cpp (hunk 4 above)
[the remark this answers is not in the document]
Comment (rplzzz) [reply source/components/n2o_component.cpp]: > A foolish consistency is the hobgoblin of little minds, adored by little statesmen and philosophers and divines. With consistency a great soul has simply nothing to do. He may as well concern himself with his shadow on the wall.
   --Ralph Waldo Emerson, _Self Reliance_

> Consistency is contrary to nature, contrary to life. The only completely consistent people are the dead.
  -- Aldous Huxley, _Do What You Will_

> Consistency is the last refuge of the unimaginative.
  -- Oscar Wilde, "The Relation of Dress to Art"

> Consistency is contrary to nature, contrary to life. The only completely consistent people are the dead.
  -- Aldous Huxley, _Do What You Will_

> Consistency is the last refuge of the unimaginative.
  -- George Bernard Shaw, quoted in _Bernard Shaw: The Lure of Fantasy_

Who are we to argue with those luminaries?  

(To be clear, I'm not saying you should go back and change it; I just don't see any reason for you (or me) to write code that we consider to be ugly just because someone else did.)
